src/Files.App/Views/Properties/HashesPage.xaml.cs

```diff
@@ -46,6 +48,75 @@ private void ToggleMenuFlyoutItem_Click(object sender, RoutedEventArgs e)
 			_cancel = true;
 		}
 
+		private void HashInputTextBox_TextChanged(object sender, TextChangedEventArgs e)
+			=> CompareHashes();
+
+		private void CompareHashes()
+		{
+			var hashToCompare = HashInputTextBox.Text;
+
+			if (string.IsNullOrEmpty(hashToCompare))
+			{
+				HashMatchInfoBar.IsOpen = false;
+				return;
+			}
+
+			foreach (var hashInfo in HashesViewModel.Hashes)
+			{
+				if (hashInfo.HashValue != null && hashInfo.HashValue.Equals(hashToCompare, StringComparison.OrdinalIgnoreCase))
+				{
+					HashMatchInfoBar.Severity = InfoBarSeverity.Success; // Set success
+					HashMatchInfoBar.Title = string.Format(Strings.HashesMatch.GetLocalizedResource(), hashInfo.Algorithm);
+					HashMatchInfoBar.IsOpen = true;
+					return;
+				}
+			}
+
+			HashMatchInfoBar.Severity = InfoBarSeverity.Error; // Set error
+			HashMatchInfoBar.Title = Strings.HashesDoNotMatch.GetLocalizedResource();
+			HashMatchInfoBar.IsOpen = true;
+		}
+
+		private async void CompareFileButton_Click(object sender, RoutedEventArgs e)
+		{
+			var picker = new Windows.Storage.Pickers.FileOpenPicker();
+			picker.SuggestedStartLocation = Windows.Storage.Pickers.PickerLocationId.Desktop;
+			picker.FileTypeFilter.Add("*");
+			WinRT.Interop.InitializeWithWindow.Initialize(picker, MainWindow.Instance.WindowHandle);
+
+			var file = await picker.PickSingleFileAsync();
+			if (file != null)
+			{
+				var selectedFileHash = await CalculateSHA384HashAsync(file.Path);
+				var currentFileHash = HashesViewModel.Hashes.FirstOrDefault(h => h.Algorithm == "SHA384")?.HashValue;
+				HashInputTextBox.Text = selectedFileHash;
+				if (selectedFileHash == currentFileHash)
+				{
+					HashMatchInfoBar.Severity = InfoBarSeverity.Success; // Check mark
+					HashMatchInfoBar.Title = Strings.HashesMatch.GetLocalizedResource();
+				}
+				else
+				{
+					HashMatchInfoBar.Severity = InfoBarSeverity.Error; // Cross mark
+					HashMatchInfoBar.Title = Strings.HashesMatch.GetLocalizedResource();
+				}
+
+				HashMatchInfoBar.IsOpen = true;
+			}
+		}
+
+		private async Task<string> CalculateSHA384HashAsync(string filePath)
```

Comment:
We should be using the quickest possible calculation.

Reply:
SHA1?

Reply:
Probably MD5